tools/importer-rest-api-specs/internal/components/apidefinitions/parser/parsingcontext/build.go

```diff
@@ -0,0 +1,330 @@
+func resolveRefsForModel(ref spec.Ref, filePath, topLevelDir string, doc *loads.Document, refs map[string]spec.Schema) error {
+	// ignore parameters as we don't use these. we might need to include these at some point, if the go-openapi parser
+	// is unable to resolve them internally (which has not happened yet)
+	if strings.HasPrefix(ref.GetURL().Fragment, "/parameters/") {
+		return nil
+	}
+
+	// determine the model name, and a swagger doc where it should be defined
+	modelName, modelFilePath := modelNamePathFromRef(ref, filePath, topLevelDir)
+
+	// the path in the ref will no longer be needed, as we are going to resolve it ourselves
+	// we can do this because `ReferenceURL` is a *url.URL, so this will thankfully propagate upstream
+	ref.Ref.ReferenceURL.Path = ""
+
+	if _, ok := refs[modelName]; ok {
+		// skip if this has already been resolved
+		return nil
+	}
+
+	if modelFilePath == "" {
+		// no valid swagger doc was found for this model
+		return fmt.Errorf("swagger document not found for ref: %s", ref.String())
+	}
+
+	// decide whether to look for the model definition in the current file, or another referenced file
+	refDoc := doc
+	if modelFilePath != filePath {
+		newDoc, err := loads.Spec(modelFilePath)
+		if err != nil {
+			return fmt.Errorf("load swagger %s: %+v", modelFilePath, err)
+		}
+
+		refDoc = newDoc
+	}
+
+	// retrieve the resolved model
+	resolvedModel, ok := refDoc.Spec().Definitions[modelName]
+	if !ok {
+		return fmt.Errorf("resolve ref %s from %s: model not found", ref.String(), filePath)
+	}
+	refs[modelName] = resolvedModel
+
+	// look for any refs in the properties (fields) of the resolved model
+	if err := resolveRefsForProperties(resolvedModel.Properties, modelFilePath, topLevelDir, refDoc, refs); err != nil {
+		return err
+	}
+
+	// some models use `additionalProperties`, these may contain refs
+	if resolvedModel.AdditionalProperties != nil && resolvedModel.AdditionalProperties.Schema != nil {
+		if resolvedModel.AdditionalProperties.Schema.Ref.String() != "" {
+			if err := resolveRefsForModel(resolvedModel.AdditionalProperties.Schema.Ref, modelFilePath, topLevelDir, refDoc, refs); err != nil {
+				return err
+			}
+		}
+	}
+
+	// some models use a top-level `items`, i.e. in the case of a type array(object)
+	if items := resolvedModel.Items; items != nil && items.Schema != nil && items.Schema.Ref.String() != "" {
+		if err := resolveRefsForModel(items.Schema.Ref, modelFilePath, topLevelDir, refDoc, refs); err != nil {
+			return err
+		}
+	}
+
+	// look for parent models - this could be a plain child model, or a discriminated implementation
+	for _, allOf := range resolvedModel.AllOf {
+		if allOf.Ref.String() != "" {
+			if err := resolveRefsForModel(allOf.Ref, modelFilePath, topLevelDir, refDoc, refs); err != nil {
+				return err
+			}
+		}
+	}
+
+	// if this is a parent model, look for implementations. this will currently only work for children defined in
+	// the same swagger file.
```

Comment:
Do we handle the case where the children are defined in a different swagger file somewhere else in `build.go`?

Context: https://github.com/hashicorp/pandora/pull/3727

Reply:
Yes, this should get picked up by https://github.com/hashicorp/pandora/pull/4307/files#diff-d2cb55dda7e9facde77a39139c64afb00da29f5f92150f1e9eec14766b11c4b1 (which I think is your original implementation)